packages/s3-deploy/lib/S3Ops.mjs

```diff
@@ -109,13 +112,48 @@ class S3Ops {
   }
 
   remoteRelativeToUri(uri) {
-    if( !uri ) return;
-    if( uri.length > 5 && uri.substring(0,5)==='s3://' ) return uri;
-    return this.group.path ? 
+    if (!uri) return;
+    if (uri.length > 5 && uri.substring(0, 5) === 's3://') return uri;
+    return this.group.path ?
       `s3://${this.group.Bucket}${this.group.path}/${uri}` :
       `s3://${this.group.Bucket}/${uri}`;
   }
 
+  checkExclude(Key, fileName, excludeExisting) {
+    return this.shouldSkip(excludeExisting[Key], fileName);
+  }
+
+  shouldSkip(item, fileName) {
+    if (!item) return false;
+    if (!fs.existsSync(fileName)) {
+      console.verbose("Doesn't exist, not skipping", fileName);
+      return false;
+    }
+    const info = fs.statSync(fileName);
+    if (item.size !== info.size) {
+      console.verbose("Size different", item.size, info.size);
+      return false;
+    }
+    // Files larger than a meg are compared ONLY on size
+    if (info.size > 1024 * 1024) return true;
```

Comment:
> is this anything related to execFileSync terminate for >1MB? (maxBuffer) https://nodejs.org/api/child_process.html#child_processexecfilesyncfile-args-options

No, nothing at all, it is an optimization to avoid running long running commands on big files.